### PR TITLE
Fix for `Get-Content -ReadCount 0` behavior when `-TotalCount` is set

### LINKED ISSUE #10707: Get-Content -ReadCount 0 does not respect -TotalCount
`Get-Content -ReadCount 0` is a convenient way to request that all lines be read _at once, into an array_ and to have that array be output _as a single_ object to the success stream.
If the intent is to read all lines into memory anyway, this can greatly speed up the command.

`-TotalCount <n>` allows you to limit the overall number of lines to be read from the file to a given number.

It's reasonable to expect that `-ReadCount 0` also respects that number and therefore returns an `<n>`-element array.

Currently, `-ReadCount 0` effectively _ignores_ `-TotalCount` and always returns _all_ lines in the file as an array.

By contrast, a specific number - e.g, `-ReadCount 2` to read the file as a stream of 2-element arrays - _does_ respect `-TotalCount`.

# Steps to reproduce

```powershell
1..10 > temp:/$pid.txt

(Get-Content temp:/$pid.txt -TotalCount 3 -ReadCount 0).Count | Should -Be 3

Remove-Item temp:/$pid.txt
```

# Expected behavior

The test should pass.

# Actual behavior

The test fails:

```none
Expected 3, but got 10.
```

That is, all 10 lines were read, even though only 3 were requested with `-TotalCount`

# Environment data

```none
PowerShell Core 7.0.0-preview.4
```

### PR DESCRIPTION
**Pull request by eugenesmlv**
# PR Summary

`-ReadCount 0` parameter of `Get-Count` command is now respect `-TotalCount <n>` parameter

## PR Context

Fix #10707 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by vexx32 — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 [hunk 1]
```diff
@@ -342,6 +342,12 @@ baz
             $expected[1] = $thirdLine
             Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should -BeNullOrEmpty
         }
+        It "Should return first three lines at a time for -TotalCount 3 and -ReadCount 0" {
+            $result = Get-Content -Path $testPath -TotalCount 3 -ReadCount 0
+            $result.Length | Should -Be 3
+            $expected = $firstLine,$secondLine,$thirdLine
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: @vexx32 "Single block" comes from -Raw.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 [hunk 2]
```diff
@@ -342,6 +342,12 @@ baz
             $expected[1] = $thirdLine
             Compare-Object -ReferenceObject $expected -DifferenceObject $result | Should -BeNullOrEmpty
         }
+        It "Should return first three lines at a time for -TotalCount 3 and -ReadCount 0" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: The name should be more generic (than using 3).

Comment (eugenesmlv) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: What about "Should return the same number of first lines which set in -TotalCount at one time for -ReadCount 0" Or "Should return first n lines at a time for -TotalCount <n> and -ReadCount 0" looks better?

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (eugenesmlv) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: @vexx32 if I understood correctly `-ReadCount 0` reads all content at once into a single array so we should compare objects. Also, should I use `-Encoding` parameter of `Get-Count` in this test?

[the remark this answers is not in the document]
Comment (vexx32) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: Ah, my misunderstanding. Good to know, thank you!

Hm. I don't think it's strictly necessary to test -Encoding at the same time?

[the remark this answers is not in the document]
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: @vexx32 Do you mean that we have gaps in Encoding tests? If so please open new tracking issue.

**Review by iSazonov — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (eugenesmlv) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: @vexx32 Sorry if I was not clear enough. What I meant to say was whether I should set the `-Encoding` parameter to UTF7, as is done in other similar tests, or not. The test passes anyway, I just want to follow the coding style.

**Review by vexx32 — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@eugenesmlv Thank you for your contribution!

**Pull request merged**